contracts/cw3-dao/src/contract.rs

```diff
@@ -143,9 +143,9 @@ pub fn instantiate(
                 admin: Some(env.contract.address.to_string()),
                 label,
                 msg: to_binary(&stake_cw20::msg::InstantiateMsg {
-                    admin: Some(env.contract.address),
+                    admin: Some(env.contract.address.to_string()),
```

Comment:
`to_string()` always creates a clone. You can use `env.contract.address.into()` instead, which is a free conversion from Addr to String.